packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2228,9 +2245,6 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         if ((member: any).abstract) {
           this.raise(member.start, TSErrors.IndexSignatureHasAbstract);
         }
-        if (isStatic) {
-          this.raise(member.start, TSErrors.IndexSignatureHasStatic);
```

Comment:
Maybe `IndexSignatureHasStatic` will be never used. So we can remove it from `TSErrors`.